plugins/woocommerce/src/Admin/Features/ProductBlockEditor/ProductTemplate.php

```diff
@@ -171,6 +182,15 @@ public function get_icon() {
 		return $this->icon;
 	}
 
+	/**
+	 * Get the match function.
+	 *
+	 * @return function The match function.
+	 */
+	public function get_match_fn() {
+		return $this->match_fn;
+	}
+
```

Comment:
We may want to add a warning in this class that a user should either only use `product_data` or `match_fn`. Unless there is something else we need the `product_data` for?

Reply:
It is actually used, when we switch to a different product template in the front-end, `product_data` is the data that is added to the product. So they can co-exist, but match_fn is used for matching if it's there.